diesel_compile_tests/tests/compile-fail/right_side_of_left_join_requires_nullable.rs

```diff
@@ -18,26 +18,122 @@ table! {
     }
 }
 
+table! {
+    pets {
+        id -> Integer,
+        user_id -> Integer,
+        name -> Text,
+    }
+}
+
 joinable!(posts -> users (user_id));
-allow_tables_to_appear_in_same_query!(posts, users);
+joinable!(pets -> users (user_id));
+allow_tables_to_appear_in_same_query!(posts, users, pets);
 sql_function!(fn lower(x: Text) -> Text);
 
 fn main() {
-    let conn = PgConnection::establish("some url").unwrap();
+}
+
+fn direct_joins() {
     let join = users::table.left_outer_join(posts::table);
 
     // Invalid, only Nullable<title> is selectable
     let _ = join.select(posts::title);
     //~^ ERROR E0271
+    //~| ERROR E0277
+    // Valid
+    let _ = join.select(posts::title.nullable());
+    // Valid -- NULL to a function will return null
+    let _ = join.select(lower(posts::title).nullable());
+    // Invalid, only Nullable<title> is selectable
+    let _ = join.select(lower(posts::title));
+    //~^ ERROR E0271
+    //~| ERROR E0277
+    // Invalid, Nullable<title> is selectable, but lower expects not-null
+    let _ = join.select(lower(posts::title.nullable()));
+    //~^ ERROR E0271
     //~| ERROR E0271
+}
+
+fn nested_outer_joins_left_associative() {
+
+    let join = users::table.left_outer_join(posts::table).left_outer_join(pets::table);
+
+    // Invalid, only Nullable<title> is selectable
+    let _ = join.select(posts::title);
+    //~^ ERROR E0271
+    //~| ERROR E0277
     // Valid
     let _ = join.select(posts::title.nullable());
     // Valid -- NULL to a function will return null
     let _ = join.select(lower(posts::title).nullable());
     // Invalid, only Nullable<title> is selectable
     let _ = join.select(lower(posts::title));
     //~^ ERROR E0271
+    //~| ERROR E0277
+    // Invalid, Nullable<title> is selectable, but lower expects not-null
+    let _ = join.select(lower(posts::title.nullable()));
+    //~^ ERROR E0271
     //~| ERROR E0271
```

Comment:
We should probably add one test for `pets` here? Just to be sure…
(Also for the other tests below?)

Reply:
We've got a lot of tests for the right side of the outermost join in this file and elsewhere. We can always add more, but they'd be redundant

Reply:
Oh right, that's the outer type wrapper in the final type. I've missed that. So just ignore this comment.